resources/iam/resource_machine_user.go

```diff
@@ -0,0 +1,264 @@
+			MinLifetimeDate: types.StringValue(mu.WorkloadPasswordDetails.PasswordMinLifetimeDate.String()),
+		})
+	if mu.AzureCloudIdentities != nil {
+		aci := make([]azureCloudIdentity, 0, len(mu.AzureCloudIdentities))
+		for _, v := range mu.AzureCloudIdentities {
+			aci = append(aci, azureCloudIdentity{
+				EnvironmentCrn: types.StringValue(v.EnvironmentCrn),
+				ObjectId:       types.StringPointerValue(v.ObjectID),
+			})
+		}
+		data.AzureCloudIdentities, _ = types.SetValueFrom(ctx, types.ObjectType{
+			AttrTypes: map[string]attr.Type{
+				"environment_crn": types.StringType,
+				"object_id":       types.StringType,
+			},
+		}, aci)
+	}
+}
+
+func (r *machineUserResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data machineUserResourceModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read API call logic
+	client := r.client.Iam
+
+	params := operations.NewListMachineUsersParamsWithContext(ctx)
+	params.WithInput(&models.ListMachineUsersRequest{
+		MachineUserNames: []string{data.Name.ValueString()},
+	})
+
+	responseOk, err := client.Operations.ListMachineUsers(params)
+	if err != nil {
+		utils.AddIamDiagnosticsError(err, &resp.Diagnostics, "read Machine User")
+		if d, ok := err.(*operations.CreateMachineUserDefault); ok && d.GetPayload() != nil && d.GetPayload().Code == "NOT_FOUND" {
+			resp.Diagnostics.AddWarning("Resource not found on provider", "Machine User not found, removing from state.")
+			tflog.Warn(ctx, "Machine User not found, removing from state", map[string]interface{}{
+				"id": data.Id,
+			})
+			resp.State.RemoveResource(ctx)
+		}
+		return
+	}
+
+	if len(responseOk.Payload.MachineUsers) == 1 && responseOk.Payload.MachineUsers[0] != nil {
+		mu := responseOk.Payload.MachineUsers[0]
+		muRespToModel(ctx, mu, &data)
+
+		// Save data into Terraform state
+		resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	} else {
+		if d, ok := err.(*operations.CreateMachineUserDefault); ok && d.GetPayload() != nil && d.GetPayload().Code == "NOT_FOUND" {
+			resp.Diagnostics.AddWarning("Resource not found on provider", "Machine User not found, removing from state.")
+			tflog.Warn(ctx, "Machine User not found, removing from state", map[string]interface{}{
+				"id": data.Id,
+			})
+			resp.State.RemoveResource(ctx)
+			return
+		}
+	}
+}
+
+func (r *machineUserResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data machineUserResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Update API call logic
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
```

Comment:
seems like we don't do anything here. is this a supported operation just the operation logic is missing or we don't support updates and we should throw an error in case of calling this?